### PR TITLE
Experiment 3.1

### PR DESCRIPTION
**Pull request by vox-vulgaris**
**models.py:**
Define a new class AnimalClassifierHead in models.py for binary classification (0 = non-animal, 1 = animal).

**pre_training.py:**
Modify the run_dicts in pre_training.py to include the AnimalClassifierHead in relevant configurations and switch to create_mixed_dataloaders. We do this by 1. importing AnimalClassifierHead, 2. updating run_dicts loop to append the new head and its loss/eval functions, 3. changing the dataloader creation to use create_mixed_dataloaders.

**data.py:**
Modify the OxfordPetDataset class in data.py to provide the "is_animal" target, returning 1 for all animal images.

**mixed_data.py:**
Modify the BackgroundDataset class in mixed_data.py to set "is_animal": 0 for non-animal images.

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -721,3 +722,47 @@ def visualize_cam_samples(dataloader, num_samples=4):
 
         plt.tight_layout()
         plt.show()
+
+# Via binary classification, enables model to explicitly distinguish between animal and non-animal images
+# This would hopefully improve CAMs by better identifying animal regions
+class AnimalClassifierHead(nn.Module):
```
Comment (timfarkas) [inline models.py:728]: instead of this you can just use the regular classifier head with num_classes = 2

**Review by timfarkas — APPROVED** [review]
